src/ngraph/op/product.hpp

```diff
@@ -23,36 +23,90 @@ namespace ngraph
-        class Product : public util::ArithmeticReduction
+        namespace v0
         {
-        public:
-            NGRAPH_API
-            static const std::string type_name;
-            const std::string& description() const override { return type_name; }
-            /// \brief Constructs a product reduction operation.
-            Product() = default;
-            /// \brief Constructs a product reduction operation.
+            /// \brief Product reduction operation.
             ///
-            /// \param arg The tensor to be reduced.
-            /// \param reduction_axes The axis positions (0-based) to be eliminated.
-            Product(const Output<Node>& arg, const AxisSet& reduction_axes);
-            /// \brief Constructs a product reduction operation.
-            ///
-            /// \param arg The tensor to be reduced.
-            /// \param reduction_axes The axis positions (0-based) to be eliminated.
-            Product(const Output<Node>& arg, const Output<Node>& reduction_axes);
+            /// Reduces the tensor, eliminating the specified reduction axes by taking the product.
+            class Product : public util::ArithmeticReduction
+            {
+            public:
+                NGRAPH_API
+                static const std::string type_name;
+                const std::string& description() const override { return type_name; }
+                /// \brief Constructs a product reduction operation.
+                Product() = default;
+                /// \brief Constructs a product reduction operation.
+                ///
+                /// \param arg The tensor to be reduced.
+                /// \param reduction_axes The axis positions (0-based) to be eliminated.
+                Product(const Output<Node>& arg, const AxisSet& reduction_axes);
+                /// \brief Constructs a product reduction operation.
+                ///
+                /// \param arg The tensor to be reduced.
+                /// \param reduction_axes The axis positions (0-based) to be eliminated.
+                Product(const Output<Node>& arg, const Output<Node>& reduction_axes);
+
+                /// \return The default value for Product.
+                virtual std::shared_ptr<Node> get_default_value() const override
+                {
+                    return ngraph::make_constant_from_string("1", get_element_type(), get_shape());
+                }
 
-            /// \return The default value for Product.
-            virtual std::shared_ptr<Node> get_default_value() const override
+                virtual std::shared_ptr<Node>
+                    copy_with_new_args(const NodeVector& new_args) const override;
+            };
+        }
+
+        namespace v1
+        {
+            /// \brief Product reduction operation.
+            ///
+            /// Reduces the tensor, eliminating the specified reduction axes by taking the product.
+            class ReduceProd : public util::ArithmeticReduction
```

Comment:
From my point of view we should add a new cpp/hpp files with new name. It looks inconvenient that product.hpp contains ReduceProd class.

Reply:
I agree with you that it is a bit incompatible with correct code style but by now (as I know) it is only one case when name of ops has been changed between opsets versions. I have no clear attitude to your proposition. @aprocter, @diyessi What do you think about it?

Reply:
I agree with @ilyachur here, but I'd be open to fixing it in a later PR.

Reply:
Done. `ReduceProd` and `ReduceSum` were placed to separated files.

Reply:
Iliya's comment was addressed in https://github.com/NervanaSystems/ngraph/pull/3556/commits/82b6389c3f0a016d7963fa22e297e0a82dd097c2